Assets/Scripts/Platforms/PlatformCollision.cs

```diff
@@ -0,0 +1,13 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+public class PlatformCollision: MonoBehaviour
```

Comment:
Hold on, I don't see this component on any of the platforms in crater town, is it not in use?

Reply:
Oh nevermind it's on the one that moves up and down – so it isn't required on the others?

Reply:
It is attached to a child of the platform named "Platform". The purpose of the child is to vertically reverse the movement of the platform when colliding with a player that is below the platform.

Reply:
And thus it isn't necessary for platforms that move horizontally? I'd imagine there could be cases where platforms should reverse horizontally as well, so renaming it to something _vertical_ doesn't seem necessary. However, perhaps it could be renamed to reflect its bounce-back intent? (the platforms already have collisions)

Reply:
I can rename it, but do you think that it is realistic that a player is stomped by a platform that moves horizontally? All platforms does reverse when reaching the end, but the purpose of the script is to reverse stomping players/avoid unintended interactions.

Reply:
Nonono that's not what I meant :sweat_smile: It's just that I only saw this component on the vertically moving platform and started thinking about whether it was general or not. I _do_ think that players should not be stomped by platforms, and that if there's a horizontal platform that closes in on a player that is falling down in the gap between the platform and its destination, we should avoid potential shenanigans by reversing it. The name sounds better now :+1: